### PR TITLE
MESH-1598/ use expect/assert in benchmarks

### PR DESCRIPTION
**Pull request by JMoore96**
Changed all `?` operators in the benchmarks to `unwrap()` and changed all `ensure`'s to `assert`.

### CODE REVIEW

File: pallets/common/src/benchs/asset.rs [hunk 1]
```diff
@@ -82,7 +86,8 @@ where
         vec![],
         None,
     )
-    .map_err(|_| "Asset cannot be created")?;
+    .map_err(|_| "Asset cannot be created")
+    .unwrap();
```
Comment (Centril) [inline pallets/common/src/benchs/asset.rs:90]:
```suggestion
    .unwrap();
```

File: pallets/compliance-manager/src/benchmarking.rs [hunk 2]
```diff
@@ -185,8 +185,8 @@ benchmarks! {
     }: _(d.owner.origin, d.ticker, d.sender_conditions.clone(), d.receiver_conditions.clone())
     verify {
         let req = Module::<T>::asset_compliance(d.ticker).requirements.pop().unwrap();
-        ensure!( req.sender_conditions == d.sender_conditions, "Sender conditions not expected");
-        ensure!( req.receiver_conditions == d.receiver_conditions, "Sender conditions not expected");
+        assert!( req.sender_conditions == d.sender_conditions, "Sender conditions not expected");
+        assert!( req.receiver_conditions == d.receiver_conditions, "Sender conditions not expected");
```
Comment (Centril) [inline pallets/compliance-manager/src/benchmarking.rs:189]:
```suggestion
        assert_eq!(req.sender_conditions, d.sender_conditions, "Sender conditions not expected");
        assert_eq!(req.receiver_conditions, d.receiver_conditions, "Sender conditions not expected");
```

File: pallets/compliance-manager/src/benchmarking.rs [hunk 3]
```diff
@@ -218,7 +218,7 @@ benchmarks! {
             .add_compliance_requirement().build();
     }: _(d.owner.origin, d.ticker)
     verify {
-        ensure!( Module::<T>::asset_compliance(d.ticker).paused == true, "Asset compliance is not paused");
+        assert!( Module::<T>::asset_compliance(d.ticker).paused == true, "Asset compliance is not paused");
```
Comment (Centril) [inline pallets/compliance-manager/src/benchmarking.rs:221]:
```suggestion
        assert!(Module::<T>::asset_compliance(d.ticker).paused, "Asset compliance is not paused");
```

File: pallets/compliance-manager/src/benchmarking.rs [hunk 4]
```diff
@@ -230,7 +230,7 @@ benchmarks! {
             d.ticker.clone()).unwrap();
     }: _(d.owner.origin, d.ticker)
     verify {
-        ensure!( Module::<T>::asset_compliance(d.ticker).paused == false, "Asset compliance is paused");
+        assert!( Module::<T>::asset_compliance(d.ticker).paused == false, "Asset compliance is paused");
```
Comment (Centril) [inline pallets/compliance-manager/src/benchmarking.rs:233]:
```suggestion
        assert!(!Module::<T>::asset_compliance(d.ticker).paused, "Asset compliance is paused");
```

File: pallets/compliance-manager/src/benchmarking.rs [hunk 5]
```diff
@@ -290,7 +290,7 @@ benchmarks! {
             .into_iter()
             .find(|req| req.id == new_req.id)
             .unwrap();
-        ensure!( req == new_req,
+        assert!( req == new_req,
```
Comment (Centril) [inline pallets/compliance-manager/src/benchmarking.rs:293]:
```suggestion
        assert_eq!(req, new_req,
```

File: pallets/compliance-manager/src/benchmarking.rs [hunk 6]
```diff
@@ -327,7 +327,7 @@ benchmarks! {
     }: _(d.owner.origin, d.ticker, asset_compliance.clone())
     verify {
         let reqs = Module::<T>::asset_compliance(d.ticker).requirements;
-        ensure!( reqs == asset_compliance, "Asset compliance was not replaced");
+        assert!( reqs == asset_compliance, "Asset compliance was not replaced");
```
Comment (Centril) [inline pallets/compliance-manager/src/benchmarking.rs:330]:
```suggestion
        assert_eq!(reqs, asset_compliance, "Asset compliance was not replaced");
```

File: pallets/contracts/src/benchmarking.rs [hunk 7]
```diff
@@ -171,76 +172,76 @@ benchmarks! {
...
     // No catalyst.
     change_template_fees {
         let creator = UserBuilder::<T>::default().generate_did().build("creator");
-        let code_hash = emulate_blueprint_in_storage::<T>(100, creator.origin.clone(), "")?;
+        let code_hash = emulate_blueprint_in_storage::<T>(100, creator.origin.clone(), "").unwrap();
     }: _(creator.origin, code_hash, Some(500u32.into()), Some(650u32.into()))
     verify {
-        ensure!(Module::<T>::get_template_details(code_hash).get_instantiation_fee() == 500u32.into(), "Contracts_change_template_fees: Failed to change the instantiation fees");
-        ensure!(Module::<T>::get_metadata_of(code_hash).usage_fee == 650u32.into(), "Contracts_change_template_fees: Failed to change the usage fees");
+        assert!(Module::<T>::get_template_details(code_hash).get_instantiation_fee() == 500u32.into(), "Contracts_change_template_fees: Failed to change the instantiation fees");
```
Comment (Centril) [inline pallets/contracts/src/benchmarking.rs:232]:
```suggestion
        assert_eq!(Module::<T>::get_template_details(code_hash).get_instantiation_fee(), 500u32.into(), "Contracts_change_template_fees: Failed to change the instantiation fees");
```

File: pallets/contracts/src/benchmarking.rs [hunk 8]
```diff
@@ -171,76 +172,76 @@ benchmarks! {
...
+        assert!(Module::<T>::get_metadata_of(code_hash).usage_fee == 650u32.into(), "Contracts_change_template_fees: Failed to change the usage fees");
```
Comment (Centril) [inline pallets/contracts/src/benchmarking.rs:233]:
```suggestion
        assert_eq!(Module::<T>::get_metadata_of(code_hash).usage_fee, 650u32.into(), "Contracts_change_template_fees: Failed to change the usage fees");
```

File: pallets/contracts/src/benchmarking.rs [hunk 9]
```diff
@@ -171,76 +172,76 @@ benchmarks! {
...
         // Catalyst for the MetaUrl length.
         let u in 1 .. MAX_URL_LENGTH;
         let url = Some(MetaUrl::from(vec![b'U'; u as usize].as_slice()));
         let creator = UserBuilder::<T>::default().generate_did().build("creator");
-        let code_hash = emulate_blueprint_in_storage::<T>(100, creator.origin.clone(), "")?;
+        let code_hash = emulate_blueprint_in_storage::<T>(100, creator.origin.clone(), "").unwrap();
     }: _(creator.origin, code_hash, url.clone())
     verify {
-        ensure!(Module::<T>::get_metadata_of(code_hash).url == url, "Contracts_change_template_meta_url: Failed to change the url of template");
+        assert!(Module::<T>::get_metadata_of(code_hash).url == url, "Contracts_change_template_meta_url: Failed to change the url of template");
```
Comment (Centril) [inline pallets/contracts/src/benchmarking.rs:244]:
```suggestion
        assert_eq!(Module::<T>::get_metadata_of(code_hash).url, url, "Contracts_change_template_meta_url: Failed to change the url of template");
```

File: pallets/contracts/src/benchmarking.rs [hunk 10]
```diff
@@ -252,9 +253,9 @@ benchmarks! {
     }: _(RawOrigin::Root, schedule)
 
     set_put_code_flag {
-        ensure!(Module::<T>::is_put_code_enabled() == false, "Contracts_set_put_code_flag: Unexpected initial state");
+        assert!(Module::<T>::is_put_code_enabled() == false, "Contracts_set_put_code_flag: Unexpected initial state");
```
Comment (Centril) [inline pallets/contracts/src/benchmarking.rs:256]:
```suggestion
        assert!(!Module::<T>::is_put_code_enabled(), "Contracts_set_put_code_flag: Unexpected initial state");
```

File: pallets/contracts/src/benchmarking.rs [hunk 11]
```diff
@@ -252,9 +253,9 @@ benchmarks! {
...
     }: _(RawOrigin::Root, true)
     verify {
-        ensure!(Module::<T>::is_put_code_enabled() == true, "Contracts_set_put_code_flag: Failed to change the flag");
+        assert!(Module::<T>::is_put_code_enabled() == true, "Contracts_set_put_code_flag: Failed to change the flag");
```
Comment (Centril) [inline pallets/contracts/src/benchmarking.rs:259]:
```suggestion
        assert!(Module::<T>::is_put_code_enabled(), "Contracts_set_put_code_flag: Failed to change the flag");
```

File: pallets/corporate-actions/src/ballot/benchmarking.rs [hunk 12]
```diff
@@ -65,7 +65,7 @@ benchmarks! {
         let (owner, ca_id) = setup_ca::<T>(CAKind::IssuerNotice);
     }: _(owner.origin(), ca_id, RANGE, meta, true)
     verify {
-        ensure!(TimeRanges::get(ca_id) == Some(RANGE), "ballot not created");
+        assert!(TimeRanges::get(ca_id) == Some(RANGE), "ballot not created");
```
Comment (Centril) [inline pallets/corporate-actions/src/ballot/benchmarking.rs:68]:
```suggestion
        assert_eq!(TimeRanges::get(ca_id), Some(RANGE), "ballot not created");
```

File: pallets/corporate-actions/src/ballot/benchmarking.rs [hunk 13]
```diff
@@ -92,14 +92,14 @@ benchmarks! {
         let results = votes.iter().map(|v| v.power).collect::<Vec<_>>();
     }: _(owner.origin(), ca_id, votes)
     verify {
-        ensure!(<Results<T>>::get(ca_id) == results, "voting results are wrong")
+        assert!(<Results<T>>::get(ca_id) == results, "voting results are wrong")
```
Comment (Centril) [inline pallets/corporate-actions/src/ballot/benchmarking.rs:95]:
```suggestion
        assert_eq!(<Results<T>>::get(ca_id), results, "voting results are wrong")
```

**Review by Centril — CHANGES_REQUESTED** [review]
These comments outline the general principle... but there are more instances of this in the code.
Please adjust the remaining ones as well.

**Review by fmiguelgarcia — APPROVED** [review]

File: pallets/asset/src/checkpoint/benchmarking.rs [hunk 14]
```diff
@@ -80,7 +80,7 @@ benchmarks! {
             },
         };
 
-        // Ensure it will fit in the max complexity.
+        // assert it will fit in the max complexity.
```
Comment (CJP10) [inline pallets/asset/src/checkpoint/benchmarking.rs:83]:
```suggestion
        // Must fit in the max complexity.
```

**Review by CJP10 — APPROVED** [review]

File: pallets/common/src/benchs/asset.rs [hunk 15]
```diff
@@ -81,7 +85,8 @@ where
...
```
Comment (Centril) [inline pallets/common/src/benchs/asset.rs:89]:
```suggestion
    .expect("Asset cannot be created");
```

File: pallets/common/src/benchs/asset.rs [hunk 16]
```diff
@@ -20,10 +20,14 @@ where
...
-        Some(name) => Ticker::try_from(name.as_ref()).map_err(|_| "Invalid ticker name")?,
+        Some(name) => Ticker::try_from(name.as_ref())
+            .map_err(|_| "Invalid ticker name")
+            .unwrap(),
         _ => Ticker::repeating(b'A'),
     };
-    Asset::register_ticker(owner, ticker).map_err(|_| "Ticker cannot be registered")?;
+    Asset::register_ticker(owner, ticker)
+        .map_err(|_| "Ticker cannot be registered")
+        .unwrap();
```
Comment (Centril) [inline pallets/common/src/benchs/asset.rs:30]:
```suggestion
        .expect("Ticker cannot be registered");
```

File: pallets/common/src/benchs/asset.rs [hunk 17]
```diff
@@ -20,10 +20,14 @@ where
     N: AsRef<[u8]>,
 {
     let ticker = match &opt_name {
-        Some(name) => Ticker::try_from(name.as_ref()).map_err(|_| "Invalid ticker name")?,
+        Some(name) => Ticker::try_from(name.as_ref())
+            .map_err(|_| "Invalid ticker name")
+            .unwrap(),
```
Comment (Centril) [inline pallets/common/src/benchs/asset.rs:25]:
```suggestion
            .expect("Invalid ticker name"),
```

File: pallets/sto/src/benchmarking.rs [hunk 18]
```diff
@@ -173,39 +177,39 @@ benchmarks! {
...
+        assert!(<Asset<T>>::balance_of(&OFFERING_TICKER, bob.user.did()) > 0u32.into(), "invest");
     }
 
     freeze_fundraiser {
-        let (alice, _) = setup_fundraiser::<T>(0, 1, 0)?;
+        let (alice, _) = setup_fundraiser::<T>(0, 1, 0).unwrap();
     }: _(alice.user.origin(), OFFERING_TICKER, 0)
     verify {
-        ensure!(<Fundraisers<T>>::get(OFFERING_TICKER, 0).unwrap().status == FundraiserStatus::Frozen, "freeze_fundraiser");
+        assert!(<Fundraisers<T>>::get(OFFERING_TICKER, 0).unwrap().status == FundraiserStatus::Frozen, "freeze_fundraiser");
```
[the remark this answers is not in the document]
Comment (JMoore96) [reply pallets/sto/src/benchmarking.rs:187]: For this change I got an error saying  `FundraiserStatus` cannot be formatted using `{:?}`, `assert_eq!(<Fundraisers<T>>::get(OFFERING_TICKER, 0).unwrap().status, FundraiserStatus::Frozen, "freeze_fundraiser");`

File: pallets/settlement/src/benchmarking.rs [hunk 19]
```diff
@@ -552,8 +553,8 @@ benchmarks! {
     }: _(origin, venue_id, Some(venue_details), Some(venue_type))
     verify {
         let updated_venue_details = Module::<T>::venue_info(1).unwrap();
-        ensure!(matches!(updated_venue_details.venue_type, VenueType::Sto), "Incorrect venue type value");
-        ensure!(matches!(updated_venue_details.details, venue_details), "Incorrect venue details");
+        assert!(matches!(updated_venue_details.venue_type, VenueType::Sto), "Incorrect venue type value");
+        assert!(matches!(updated_venue_details.details, venue_details), "Incorrect venue details");
```
[the remark this answers is not in the document]
Comment (JMoore96) [reply pallets/settlement/src/benchmarking.rs:557]: I got an error stating the ventue_details is 'value borrowed here after move' as in line 553 it is used in Some(venue_details)

File: pallets/pips/src/benchmarking.rs [hunk 20]
```diff
@@ -169,7 +169,7 @@ fn propose_verify<T: Trait>(url: Url, description: PipDescription) -> DispatchRe
 
 fn execute_verify<T: Trait>(state: ProposalState, err: &'static str) -> DispatchResult {
     if Proposals::<T>::contains_key(&0) {
-        ensure!(state == Module::<T>::proposals(&0).unwrap().state, err);
+        assert!(state == Module::<T>::proposals(&0).unwrap().state, err);
```
[the remark this answers is not in the document]
Comment (JMoore96) [reply pallets/pips/src/benchmarking.rs:173]: When implementing this it gave the message 'error: format argument must be a string literal' for the err variable and suggested  `assert_eq!(state, Module::<T>::proposals(&0).unwrap().state, "{}", err);`

File: pallets/sto/src/benchmarking.rs (hunk 18 above)
[the remark this answers is not in the document]
Comment (Centril) [reply pallets/sto/src/benchmarking.rs:187]: To fix this, take the `#[cfg_attr(feature = "std", derive(Debug))]` on `FundraiserStatus` and replace it with `#[derive(Debug)]`.

File: pallets/pips/src/benchmarking.rs (hunk 20 above)
[the remark this answers is not in the document]
Comment (Centril) [reply pallets/pips/src/benchmarking.rs:173]: Yep, use that suggestion.

File: pallets/settlement/src/benchmarking.rs (hunk 19 above)
[the remark this answers is not in the document]
Comment (Centril) [reply pallets/settlement/src/benchmarking.rs:557]: Do `let venue_details2 = venue_details.clone();` right after `let venue_details = ...;` and then use `venue_details2` here.

**Review by maxsam4 — APPROVED** [review]

**Review by Centril — APPROVED** [review]

**Pull request merged**